### PR TITLE
Fading of preprocessor statements (only instruction part)

### PR DESCRIPTION
**Pull request by jirimosinger**

### CODE REVIEW

File: parser_library/test/semantics/highlighting_test.cpp [hunk 1]
```diff
@@ -399,6 +399,8 @@ TEST(highlighting, endevor_preprocessor_statement)
     const std::string contents = R"(
 -INC  MEMBER bla bla)";
 
+    auto fms = std::make_shared<std::vector<fade_message_s>>();
```
Comment (slavek-kucera) [inline parser_library/test/semantics/highlighting_test.cpp:402]: unused?

File: parser_library/src/processing/preprocessors/cics_preprocessor.cpp [hunk 2]
```diff
@@ -1110,7 +1110,8 @@ class cics_preprocessor final : public preprocessor
         const preprocessor::line_iterator& end,
         const std::optional<size_t>& potential_lineno)
     {
-        static const std::regex exec_cics("^([^ ]*)[ ]+([eE][xX][eE][cC][ ]+[cC][iI][cC][sS](?:[ ]+(\\S+))?)(?= |$)");
+        static const std::regex exec_cics(
+            "^([^ ]*)[ ]+(?:([eE][xX][eE][cC][ ]+[cC][iI][cC][sS])(?:[ ]+(\\S+))?)(?= |$)");
```
Comment (slavek-kucera) [inline parser_library/src/processing/preprocessors/cics_preprocessor.cpp:1114]: is the first non-capture group actually needed?

**Review by slavek-kucera — APPROVED** [review]

File: parser_library/src/processing/preprocessors/cics_preprocessor.cpp (hunk 2 above)
Comment (jirimosinger) [reply parser_library/src/processing/preprocessors/cics_preprocessor.cpp:1114]: The following test doesn't work without the non-capturing group:
https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/blob/0ccd11d675fd3f4dc4c2b76a7391ffbc90cf1353/parser_library/test/processing/cics_preprocessor_test.cpp#L483

Comment (jirimosinger) [reply parser_library/src/processing/preprocessors/cics_preprocessor.cpp:1114]: Actually, seems like I removed wrong parenthesis and the test does indeed pass with this regex: 
`^([^ ]*)[ ]+([eE][xX][eE][cC][ ]+[cC][iI][cC][sS])(?:[ ]+(\\S+))?(?= |$)`

Comment (slavek-kucera) [reply parser_library/src/processing/preprocessors/cics_preprocessor.cpp:1114]: This part `(?:([eE][xX]`, not `(?:[ ]+(\\S+))?`

Comment (jirimosinger) [reply parser_library/src/processing/preprocessors/cics_preprocessor.cpp:1114]: Yup, I meant that I removed wrong _**end**_ parenthesis and didn't notice at first.

**Pull request merged**